astropy/io/fits/column.py

```diff
@@ -1429,6 +1429,12 @@ def dtype(self):
                 else:
                     dt = np.dtype((dt.base, dim))
 
+            # On Python 2, force the `name` to `str`
+            # because Numpy structured arrays can't handle unicode `name`
+            # See https://github.com/astropy/astropy/issues/5204
+            if PY2:  # pragma: py2
+                name = str(name)
```

Comment:
Actually on Python 3 there is a verification to see if the string has only ascii characters:
```
In [12]: col = fits.Column(name='spàm', format='E', array=[42.])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-12-35a74b04b819> in <module>()
----> 1 col = fits.Column(name='spàm', format='E', array=[42.])

....

/home/simon/dev/astropy/astropy/io/fits/card.py in value(self, value)
    580                     'FITS header values must contain standard printable ASCII '
    581                     'characters; %r contains characters not representable in '
--> 582                     'ASCII or non-printable characters.' % value)
    583         elif isinstance(value, binary_type):
    584             # Allow str, but only if they can be decoded to ASCII text; note

ValueError: FITS header values must contain standard printable ASCII characters; 'spàm' contains characters not representable in ASCII or non-printable characters.
```
Ideally, this verification should also be done on Python 2 to have a more informative exception. Otherwise here with `str(name)` it will raise a `UnicodeEncodeError`.
But it may be difficult to change this, as it seems that, with PY2, the code before breaks with Numpy before it goes to the verification step.